### PR TITLE
Remove second descriptor because of FP rate

### LINKED ISSUE #4432: CA2248 has incorrect and undocumented diagnostic
### Analyzer

**Diagnostic ID**: [CA2248](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2248): `Provide correct enum argument to Enum.HasFlag`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

**Version**: v3.3.1 (Latest)

### Describe the bug

The CA2248 diagnostic incorrectly fires on uses of `HasFlag` that appears on enum types that are missing a `[Flags]` attribute, claiming that "this call will always return 'false'" when in fact it *may* return true just like any other enum, as attested to by this C# Interactive Window output:

```
> enum A { B = 1
. }
> A a = A.B;
> a.HasFlag(A.B)
true
> 
```

### Steps To Reproduce

Given this code:

```cs
            if (((__VSFINDOPTIONS)grfOptions).HasFlag(__VSFINDOPTIONS.FR_Find) ||
                ((__VSFINDOPTIONS)grfOptions).HasFlag(__VSFINDOPTIONS.FR_Replace))
```

Where the enum is defined as:

```cs
public enum __VSFINDOPTIONS
{
FR_Find = 0x100000,
FR_Replace = 0x400000
}
```

### Expected behavior

No warning. In all cases, whether the enum is attributed with a `FlagsAttribute` or not, or even if its members are defined with flags-friendly values, HasFlag will indeed return `true` if the given value is set as if it _were_ a flags enum.

### Actual behavior

The analyzer incorrectly claims:
 
> This call will always returns 'false' because the enum type '__VSFINDOPTIONS' is not marked with 'FlagsAttribute'

It does _not_ return false, so the message is dangerously misleading as someone might remove the code that is guarded by the condition that allegedly will never evaluate to true.

### PR DESCRIPTION
**Pull request by Evangelink**
Fix #4432

### CODE REVIEW

**Review by mavasani — APPROVED** [review]

File: src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Usage/ProvideCorrectArgumentToEnumHasFlagTests.cs [hunk 1]
```diff
@@ -92,40 +92,6 @@ End Class
 ");
         }
 
-        [Fact]
-        public async Task CA2248_EnumTypesAreSameButNotFlag_NoDiagnostic()
```
Comment (sharwell) [inline src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Usage/ProvideCorrectArgumentToEnumHasFlagTests.cs:96]: 💡 Consider keeping this test but updating it to show the new outcome

File: src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Usage/ProvideCorrectArgumentToEnumHasFlag.cs [hunk 2]
```diff
@@ -54,26 +43,18 @@ public override void Initialize(AnalysisContext context)
...
-                        invocation.Arguments[0].Value is not IConversionOperation conversion ||
-                        invocation.Instance.Type == null ||
-                        invocation.Instance.Type.TypeKind == TypeKind.TypeParameter ||
-                        conversion.Operand.Type?.TypeKind == TypeKind.TypeParameter)
-                    {
-                        return;
-                    }
-
-                    if (!invocation.Instance.Type.Equals(conversion.Operand.Type))
+                    if (invocation.TargetMethod.ContainingType.SpecialType == SpecialType.System_Enum &&
```
Comment (sharwell) [inline src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Usage/ProvideCorrectArgumentToEnumHasFlag.cs:46]: 💭 I think I would have preferred to not change the flow control layout here, but I suppose what's done is done

**Review by sharwell — APPROVED** [review]
LGTM. If another test already covers the case I commented on, feel free to mention the test name instead of adding a new one.

File: src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Usage/ProvideCorrectArgumentToEnumHasFlag.cs (hunk 2 above)
Comment (Evangelink) [reply src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Usage/ProvideCorrectArgumentToEnumHasFlag.cs:46]: I can revert, it just seemed weird to have all the conditions for the return and then one if for the actual condition.

File: src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx [hunk 3]
```diff
@@ -1,17 +1,17 @@
...
+    
+    The primary goals of this format is to allow a simple XML format 
+    that is mostly human readable. The generation and parsing of the 
+    various data types are done through the TypeConverter classes 
     associated with the data types.
-
+    
     Example:
-
+
```
Comment (Youssef1313) [inline src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx:14]: What made these spaces to appear?

Comment (Evangelink) [reply src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx:14]: As I haven't done any manual change I guess that's VS.

Comment (Youssef1313) [reply src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx:14]: @Evangelink Not the first time I see this issue without manual changes.

@mavasani Any idea where to report this possible flaky bug?

Comment (AArnott) [reply src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx:14]: I suspect VS's resgen tool is consistently adding them, and editors configured to trim trailing whitespace remove it when the file is manually edited. 
I tend to "let VS win" in whitespace conflicts when that is the easiest thing to maintain.

Filing a suggestion bug against resgen to stop emitting the trailing whitespace in the first place would be Goodness.

**Review by AArnott — APPROVED** [review]

**Pull request merged**